rfcs/2022-02-11-responsive-and-adaptive-layout-strategy.md

```diff
@@ -0,0 +1,391 @@
+| The system can evaluate the consumers screen resolution and send the appropriate content for them hence, preventing downloading unnecessary assets required for that screen size. | Can leave out users “in the middle” if the screen sizes are not properly defined and the content is not available for a particular screen size   |
+| It offers an alternative to the “one-size-stretches-to-all” approach.                                                                                                             |                                                                                                                                                  |
+
+## Approaches taken by well-known Design Systems
+
+| Name                                                                                                    | Responsiveness handled internally/flexibility to consumers?                                                                          |
+| ------------------------------------------------------------------------------------------------------- | ------------------------------------------------------------------------------------------------------------------------------------ |
+| [Paste - Twilio](https://paste.twilio.design/primitives/text/#responsive-text)                          | Flexible with `[]` props accepting values for breakpoints on various components. Uses styled system under the hood                   |
+| [Primer - GitHub](https://primer.style/react/core-concepts#responsive-props)                            | Flexible with `[]` props accepting props accepting values for breakpoints on various components. Uses styled system under the hood   |
+| [Arcade](https://arcade.design/content/docs/tokens/typography)                                          | Handles internally, not customisable on the consumer end(nice strategy for less overhead)                                            |
+| [React Spectrum - Adobe](https://react-spectrum.adobe.com/react-spectrum/layout.html#responsive-layout) | Flexible to define breakpoints for the entire application and flexible to define behavior for those breakpoints per components basis |
+
+## What would work for us?
+
+As you saw there are pros and cons to each of the approach so we can't abandon either of them but we need to strike a strategy where each of these approaches will work best.
+
+After doing some research and aligning it with our uses cases at Razorpay, I propose following things at a broader level.
+
+### 1. Components that can handle Responsiveness Internally
+
+- Most of the components that don't deal with layouts shall handle responsiveness internally. For eg: Button, Modal, Avatar, Select, etc.
+
+- The properties that affects responsiveness are:
+
+  - `width`
+  - `height`
+  - `font-size`
+  - `padding`
+  - `margin`
+
```

Comment:
I guess `position` property might also change, right?

Reply:
Even positional values like top,left,right,bottom might be different